icu4c/source/i18n/collationbuilder.cpp

```diff
@@ -734,12 +737,15 @@ CollationBuilder::addRelation(int32_t strength, const UnicodeString &prefix,
             parserErrorReason = "normalizing the relation extension";
             return;
         }
-        cesLength = dataBuilder->getCEs(nfdExtension, ces, cesLength);
+        cesLength = dataBuilder->getCEs(nfdExtension, ces, cesLength, errorCode);
         if(cesLength > Collation::MAX_EXPANSION_LENGTH) {
             errorCode = U_ILLEGAL_ARGUMENT_ERROR;
             parserErrorReason =
                 "extension string adds too many collation elements (more than 31 total)";
             return;
+        } else if (errorCode == U_INPUT_TOO_LONG_ERROR) {
```

Comment:
no need to put the if after else since the previous block will always return and no chance it will reach the if here anyway

Reply:
Same response as above, but I went ahead and made both your and Peter's suggested changes here too.